cpp/src/jni/dataset/jni_wrapper.cc

```diff
@@ -166,6 +156,126 @@ class DisposableScannerAdaptor {
   }
 };
 
+/// \brief Simple scan task implementation that is constructed directly
+/// from a record batch iterator (and its corresponding fragment).
+class SimpleIteratorTask : public arrow::dataset::ScanTask {
```

Comment:
CC @lidavidm @westonpace what is the current status of changes in cardinality for ScanTask?  is this a use that should be supported long term?

Reply:
ScanTask is going away. However, from a quick glance, what we want here is a Scanner from a RecordBatchIterator. Use the existing Scanner::FromRecordBatchReader:

https://github.com/apache/arrow/blob/8b0e04961790003abb32ef03e19a4604060e88ec/cpp/src/arrow/dataset/scanner.h#L323-L329

Reply:
Fixed in https://github.com/apache/arrow/pull/10201/commits/71cba257cfd24cd3b37b00f8e478cbceab8bc440. Thanks for adding the facility.